src/slc/java/com/lambdaherding/edi/slc/ch03/AdvancedEx2.java

```diff
@@ -0,0 +1,23 @@
+package com.lambdaherding.edi.slc.ch03;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AdvancedEx2 {
+	
+	public static Stream<String> filterOnlyUppercase(Stream<String> strings) {
+		return strings.reduce(Stream.builder(), (a, b) -> {
+			if (b.chars().noneMatch(i -> Character.isLowerCase(i))) {
+				a.add(b);
+			}
+			return a;
+		}, (a, b) -> {
+			return a;

```

Comment:
I suspect this may have trouble with parallel streams, as it ignores `b`. This can also be written more simply as `(a, b) -> a`.